classes/journal/SectionDAO.php

```diff
@@ -456,7 +451,7 @@ public function getByJournalId($journalId, $rangeInfo = null)
     public function getByContextId($journalId, $rangeInfo = null, $submittableOnly = false)
     {
         $result = $this->retrieveRange(
-            'SELECT * FROM sections WHERE journal_id = ? ' . ($submittableOnly ? ' AND editor_restricted = 0' : '') . ' ORDER BY seq',
+            'SELECT * FROM sections WHERE journal_id = ? ' . ($submittableOnly ? ' AND editor_restricted = 0 AND is_inactive = 0' : '') . ' ORDER BY seq',
```

Comment:
I would rather `is_inactive` be specifiable as a parameter (even if the default is to exclude inactive sections). If I were coding without checking, I would expect `getByContextId` to get *everything*; if that's not the case, autocomplete/hinting should guide the dev to know what to expect.

(BTW, I just merged https://github.com/pkp/pkp-lib/issues/8366 to `main` in OMP and pkp/browse; a doubled check won't cause harm, but just FYI.)

Reply:
In this case, `is_inactive` is a condition of `$submittableOnly`. This was added when inactive sections were added. So historically, `$submittableOnly` meant "can a non-editor submit to this section". 

If I were to add `?boolean $isInactive = false` as a function argument, then it would be possible to call the function in a way that undermines this logic, since even editors can not submit to inactive sections:

```php
$submittableOnly = true;
$isInactive = true;
$sections = $sectionDao->getByContextId($contextId, null, $submittableOnly, $isInactive);
```

We could fix this by changing `$submittableOnly` to `$restrictedToEditors`. But that would be changing the _intent_ of the original function argument.

Reply:
Ah, NVM, I missed this already being in a conditional covered by the calling parameters. :+1: